samples/matter/common/src/bridge/bridge_manager.h

```diff
@@ -12,19 +12,87 @@
 
 class BridgeManager {
 public:
-	void Init();
+	static constexpr uint8_t kMaxBridgedDevices = CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+	using LoadStoredBridgedDevicesCallback = CHIP_ERROR (*)();
+
+	/**
+	 * @brief Initialize BridgeManager instance.
+	 *
+	 * @param loadStoredBridgedDevicesCb callback to method capable of loading and adding bridged devices stored in
+	 * persistent storage
+	 * @return CHIP_NO_ERROR on success
+	 * @return other error code on failure
+	 */
+	CHIP_ERROR Init(LoadStoredBridgedDevicesCallback loadStoredBridgedDevicesCb);
+
 	/**
 	 * @brief Add devices which are supposed to be bridged to the Bridge Manager
 	 *
 	 * @param devices Matter devices to be bridged (the maximum size of this array may depend on the
 	 * application and can be memory constrained)
 	 * @param dataProvider data provider which is going to be bridged with Matter devices
 	 * @param deviceListSize number of Matter devices which are going to be bridged with data provider
-	 * @return CHIP_NO_ERROR in case of success, specific CHIP_ERROR code otherwise
+	 * @param devicesPairIndexes array of pointers to the indexes that will be filled with pairs' indexes
+	 * assigned by the bridge
+	 * @return CHIP_NO_ERROR on success
+	 * @return other error code on failure
 	 */
 	CHIP_ERROR AddBridgedDevices(MatterBridgedDevice *devices[], BridgedDeviceDataProvider *dataProvider,
-				     uint8_t deviceListSize);
-	CHIP_ERROR RemoveBridgedDevice(uint16_t endpoint);
+				     uint8_t deviceListSize, uint8_t devicesPairIndexes[]);
+
+	/**
+	 * @brief Add devices which are supposed to be bridged to the Bridge Manager using specific index and endpoint
+	 * id.
+	 *
+	 * @param devices Matter devices to be bridged (the maximum size of this array may depend on the
+	 * application and can be memory constrained)
+	 * @param dataProvider data provider which is going to be bridged with Matter devices
+	 * @param deviceListSize number of Matter devices which are going to be bridged with data provider
+	 * @param devicesPairIndexes array of pointers to the indexes that contain index value required to be
```

Comment:
Isn't it an array of values/indexes now? Also please check the related doxygen bits in the overloaded methods.